### ISSUE TITLE
[sklearn#34278] Publish Pyodide/WASM (pyemscripten) wheels to PyPI via PEP 783

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#339]
**Mirrored from [scikit-learn#34278](https://github.com/scikit-learn/scikit-learn/issues/34278)**

Original title: Publish Pyodide/WASM (pyemscripten) wheels to PyPI via PEP 783
Original labels: New Feature
Created: 2026-06-13T04:07:41Z

---

### Describe the workflow you want to enable

Hi! I would like to propose that scikit-learn should start publishing Pyodide/WASM wheels, i.e., with the [pyemscripten](https://pyodide.org/en/stable/development/abi.html) platform tag as per the accepted and implemented [PEP 783](https://peps.python.org/pep-0783/), to PyPI, for both ABIs that we currently support:
- the 2025 ABI (corresponding to [Pyodide 0.29.4](https://pyodide.org/en/0.29.4/), built with CPython 3.13 and Emscripten 4.0.9); and
- the 2026 ABI (corresponding to the [new Pyodide 314.0.0 release](https://pyodide.org/en/stable/), read our [blog post](https://blog.pyodide.org/posts/314-release/) – built with CPython 3.14 and Emscripten 5.0.3).

This follows on from the RFC for Pyodide being regarded as a _somewhat_ officially supported platform posted at https://github.com/scikit-learn/scikit-learn/issues/23727, and resolved as of June 2025 (just about a year ago!).

The rationale to build this case is also Pyodide's ever-growing maturity and stability as a platform, highlighted by the PEP's [acceptance](https://discuss.python.org/t/pep-783-emscripten-packaging/86862/104), [official Tier III CPython 3.14 support as per the active PEP 776](https://docs.python.org/3/whatsnew/3.14.html); and our various cross-compilation infrastructure, packaging, and interoperability improvements over the years; besides scikit-learn's usage of Pyodide for its JupyterLite-powered interactive documentation offerings and well-established testing as part of the CI test suite.

### Describe your proposed solution

The scikit-learn wheels are built with [cibuildwheel](https://cibuildwheel.pypa.io/) everywhere across the board. [cibuildwheel v4.1.0](https://github.com/pypa/cibuildwheel/releases/tag/v4.1.0), released on 12th June, 2026, i.e., yesterday at the time of writing, is capable of building and publishing Pyodide wheels with `CIBW_PLATFORM: pyodide` environment variable or the `--platform pyodide` command-line flag-argument pair for both the 2025 and 2026 ABIs, with all batteries included. A comprehensive set of documentation for pyodide-build is available at https://pyodide-build.readthedocs.io, thanks to @ryanking13. A release blog post for cibuildwheel v4 is also available at https://iscinumpy.dev/post/cibuildwheel-4-0-0/.

As a result, this should be a simple switch (famous last words?). It would only require meddling with the workflow files to integrate uploading such built wheels to PyPI, while ensuring that the test suite is green for both ABIs.

### Describe alternatives you've considered, if relevant

The best alternative is the status quo itself, in that scikit-learn keeps running its test suite for Pyodide without publishing wheels, and we retain its recipe at https://github.com/pyodide/pyodide-recipes. However, such a scenario would not be ideal for us, as we plan to scale down the scope of that repository as we work towards supporting the adoption of this PEP across the ecosystem: https://git

...(truncated)

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/84 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
